### PR TITLE
return permission_denied__can_apply_for_trial for trial candidate

### PR DESCRIPTION
**Pull request by goneri**
Jira Issue: <https://issues.redhat.com/browse/AAP-26581>

## Description

When the user's org is not configured yet, we return a different
error message to let the VSCode extension know the user can apply for
a Trial.

## Testing

- Switch `ANSIBLE_AI_ENABLE_ONE_CLICK_TRIAL` to `True`
- Connect with a user with a valid Org but no secret
- Call the API, you should get the `permission_denied__can_apply_for_trial` error code :partying_face: 

## Production deployment

- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

File: ansible_ai_connect/ai/api/permissions.py [hunk 1]
```diff
@@ -64,6 +94,12 @@ def has_permission(self, request, view):
         if user.rh_user_has_seat is True:
             return True
```
Comment (manstis) [inline ansible_ai_connect/ai/api/permissions.py:95]: You probably want to change this to `CONTINUE` for consistency?

L#93 and L#95 (GitHub won't let me comment on L#93)

**Review by manstis — APPROVED** [review]
LGTM :+1: 

What a tangled web of permissions we have :smile_cat: 

Fortunately the unit tests make it (sort of) clear what's going on.

**Comment by goneri** [comment]
> LGTM 👍
> 
> What a tangled web of permissions we have 😸
> 
> Fortunately the unit tests make it (sort of) clear what's going on.

Yes, we are also overdue for a clean up (TechPreview/Seat)

**Review by manstis — APPROVED** [review]
Thanks. LGTM :+1:

**Review by manstis — APPROVED** [review]
LGTM :+1:

**Pull request merged**